### ISSUE TITLE
Error when using this library in AWS Lambda functions

### ISSUE THREAD

**Issue by ahhh** [request/request#2364]
I am experiencing an error when 'requiring' the requests library using Node3.4 in an AWS Lambda function. I've done A/B testing where I have a known good piece of code that does not use this library and then it can no longer parse and load the module when I add a single line to require this library and redeploy the code to AWS Lambda.

```
Unable to import module 'MyCode': Error at Function.Module._resolveFilename (module.js:325:15) at Function.Module._load (module.js:276:25) at Module.require (module.js:353:17) at require (internal/module.js:12:17) at Object.<anonymous> (/var/task/MyCode.js:20:15) at Module._compile (module.js:409:26) at Object.Module._extensions..js (module
Unable to import module 'MyCode': Error
at Function.Module._resolveFilename (module.js:325:15)
at Function.Module._load (module.js:276:25)
at Module.require (module.js:353:17)
at require (internal/module.js:12:17) 
at Object.<anonymous> (/var/task/MyCode.js:20:15)
```

^^ This is the line that I define: "var request = require('request');"

```
at Module._compile (module.js:409:26)
at Object.Module._extensions..js (module.js:416:10)
at Module.load (module.js:343:32)
at Function.Module._load (module.js:300:12)
at Module.require (module.js:353:17)
```

**Comment by srijanmishra** [comment]
I think the issue is something else. If you just added the line to your code and deployed it without installing then it won't work. You have to package the necessary node modules required by your lambda in the deployment. 

Check this link [Creating a Deployment Package (Node.js)](http://docs.aws.amazon.com/lambda/latest/dg/nodejs-create-deployment-pkg.html)

**Comment by shenie** [comment]
In your zip file for lambda... look inside a downstream dependency `mime-db`. There should be 3 files - index.js, db.json & package.json....

I was missing `db.json` because I only included **/*.js and **/package.json.

**Comment by edent** [comment]
I'm also having this issue.

In my directory, I've run `npm install request` - that downloads requests and all its dependencies. (`mime-db` is installed without the above errors)

I zip up `index.js` and `node_modules` into a single zip file.

I upload to AWS Lambda.

I get the `Unable to import module` error.

**Comment by noor92** [comment]
@edent  

have you managed to solve this issue ?

**Comment by edent** [comment]
Nope. Sorry.

**Comment by swecker** [comment]
I was having this same issue and realized that it was because some dependent modules were installed globally on my system.  When I used `npm install` it skipped those already globally installed modules, still worked fine locally, but when zipped up and used in a Lambda function it didn't have access to those modules.

I'm not sure if this is the best approach to fixing it, but when I used `npm install --link` it added all dependent modules to the local directory (via sym-links), and I was able to zip that all up and resolve my issue.

I couldn't find any help with this online anywhere, so I figured I'd at least reply here in case this helps someone else.
